backend/src/revisions/revisions.service.ts

```diff
@@ -158,7 +164,13 @@ export class RevisionsService {
       latestRevision.content,
       newContent,
     );
-    const revision = Revision.create(newContent, patch, note, yjsStateVector);
+    const revision = Revision.create(
+      newContent,
+      patch,
+      note,
+      yjsStateVector,
+    ) as Revision;
+    updateRevisionMetadataFromContent(revision);
```

Comment:
Is it valid to create a revision without metadata?

Reply:
No. that would create inconsistency because the metadata would be out of sync with the content.

Reply:
So shouldn't we put a parameter for metadata in the create method of the revision?

Reply:
I restructured and separated the code to do this